src/backend/joanie/tests/core/api/order/test_submit_installment_payment.py

```diff
@@ -0,0 +1,571 @@
+            data=payload,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.UNPROCESSABLE_ENTITY)
+        self.assertEqual(
+            response.json(),
+            {"detail": "The order is not in failed payment state."},
+        )
+
+    def test_api_order_submit_installment_payment_order_is_in_completed_state(self):
+        """
+        Authenticated user should not be able to pay for a failed installment payment
+        if its order is in state 'completed.'.
+        """
+        user = UserFactory()
+        token = self.generate_token_from_user(user)
+        payload = {"credit_card_id": uuid.uuid4()}
+        order_completed = OrderFactory(owner=user, state=ORDER_STATE_COMPLETED)
+
+        response = self.client.post(
+            f"/api/v1.0/orders/{order_completed.id}/submit_installment_payment/",
+            data=payload,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.UNPROCESSABLE_ENTITY)
+        self.assertEqual(
+            response.json(),
+            {"detail": "The order is not in failed payment state."},
+        )
+
+    def test_api_order_submit_installment_payment_with_not_matching_credit_card_id_in_payload(
+        self,
+    ):
+        """
+        When we pass a `credit_card_id` that does not exists for the user who wants to pay
+        for a failed installment payment, it should not work and return a response
+        status code `NOT_FOUND`.
+        """
+        user = UserFactory(email="john.doe@acme.org")
+        order_with_failed_payment = OrderFactory(
+            owner=user,
+            state=ORDER_STATE_FAILED_PAYMENT,
+            payment_schedule=[
+                {
+                    "id": "1932fbc5-d971-48aa-8fee-6d637c3154a5",
+                    "amount": "200.00",
+                    "due_date": "2024-01-17",
+                    "state": PAYMENT_STATE_PAID,
+                },
+                {
+                    "id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
+                    "amount": "300.00",
+                    "due_date": "2024-02-17",
+                    "state": PAYMENT_STATE_PAID,
+                },
+                {
+                    "id": "9fcff723-7be4-4b77-87c6-2865e000f879",
+                    "amount": "300.00",
+                    "due_date": "2024-03-17",
+                    "state": PAYMENT_STATE_REFUSED,
+                },
+                {
+                    "id": "168d7e8c-a1a9-4d70-9667-853bf79e502c",
+                    "amount": "199.99",
+                    "due_date": "2024-04-17",
+                    "state": PAYMENT_STATE_PENDING,
+                },
+            ],
+        )
+        payload = {"credit_card_id": uuid.uuid4()}
```

Comment:
Instead of generating a random uuid here, you should create a real credit card but not linked to the user.